all_test.go

```diff
@@ -116,3 +120,57 @@ func rungo(t *testing.T, args ...string) {
 		t.Fatalf("%v: %v\n%s", cmd, err, output)
 	}
 }
+
+func TestExportedSymbolsHaveDocs(t *testing.T) {
+	// TODO(https://github.com/googleapis/librarian/issues/522): turn on once
+	// all existing symbols have docs.
+	t.Skip()
+
+	err := filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
+		if err != nil || d.IsDir() || !strings.HasSuffix(path, ".go") || strings.HasSuffix(path, "_test.go") {
+			return nil
+		}
+
+		fset := token.NewFileSet()
+		node, err := parser.ParseFile(fset, path, nil, parser.ParseComments)
+		if err != nil {
+			t.Errorf("failed to parse file %q: %v", path, err)
+			return nil
+		}
+
+		// Visit every top-level declaration in the file.
+		for _, decl := range node.Decls {
+			gen, ok := decl.(*ast.GenDecl)
+			if ok && (gen.Tok == token.TYPE || gen.Tok == token.CONST || gen.Tok == token.VAR) {
+				for _, spec := range gen.Specs {
+					switch s := spec.(type) {
+					case *ast.TypeSpec:
+						checkDoc(t, s.Name, s.Doc, path)
+					case *ast.ValueSpec:
+						for _, name := range s.Names {
+							checkDoc(t, name, gen.Doc, path)
+						}
+					}
+				}
+			}
+			if fn, ok := decl.(*ast.FuncDecl); ok {
+				checkDoc(t, fn.Name, fn.Doc, path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func checkDoc(t *testing.T, name *ast.Ident, doc *ast.CommentGroup, path string) {
+	t.Helper()
+	if !name.IsExported() {
+		return
+	}
+	if doc == nil {
```

Comment:
Is there something else that does the "starting with their name" part of the check (from the PR description), or is that still to be implemented?

Reply:
I deleted it after this discussion: https://github.com/googleapis/librarian/pull/523#discussion_r2160513755. The goal is for this to be useful, so I'd rather have us miss something than have a check be noisy.

(thanks for the comment, I didn't catch that it was still in the commit message)